enumerables.rb

```diff
@@ -0,0 +1,115 @@
+module Enumerable
+  def my_each
+    position = 0
+    return to_enum unless block_given?
+
+    while position <= length
+      yield self[position]
+      position += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    position = 0
+    return to_enum unless block_given?
+
+    while position <= length
+      yield self[position], position
+      position += 1
+    end
+    self
+  end
+
+  # Define my_select
+  def my_select
+    result = []
+    return to_enum unless block_given?
+
+    my_each do |item|
+      result << item if yield item
+    end
+    result
+  end
+
+  def my_all?(arg = nil)
+    array = *self
+    result = true
+    if block_given?
+      array.my_each { |x| result = false unless yield(x) }
+    elsif arg.nil?
+      array.my_each { |x| result = false unless x }
+    else
+      array.my_each { |x| result = false unless arg == x }
+    end
+    result
+  end
+
+  def my_any?(arg = nil)
+    array = *self
+    result = false
+    if block_given?
+      array.my_each { |x| result = true if yield(x) }
+    elsif arg.nil?
+      array.my_each { |x| result = true if x }
+    else
+      array.my_each { |x| result = true if arg == x }
+    end
+    result
+  end
```

Comment:
Please check your method when a block is given. It's not working right now. Should return true if it never returns false or nil.
Should return an answer if you give a class as an argument.
Should return an answer if you give a regex as an argument.